### PR TITLE
Rares/webhook UI changes

### PR DESCRIPTION
**Pull request by teodosii**
# What this PR does

## Checklist

- [ ] Unit, integration, and e2e (if applicable) tests updated
- [ ] Documentation added (or `pr:no public docs` PR label added if not required)
- [ ] `CHANGELOG.md` updated (or `pr:no changelog` PR label added if not required)

### CODE REVIEW

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx [hunk 1]
```diff
@@ -19,28 +24,46 @@ const cx = cn.bind(styles);
...
+
+  if (
+    (action === WebhookFormActionType.EDIT_SETTINGS || action === WebhookFormActionType.VIEW_LAST_RUN) &&
+    !outgoingWebhook2Store.items[id]
+  ) {
+    return null;
+  }
 
   const data =
     id === 'new'
```
Comment (maskin25) [inline grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:58]: it's confusing a bit, what the difference between `action === WebhookFormActionType.NEW` and `id === 'new'`

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx [hunk 2]
```diff
@@ -49,35 +72,167 @@ const OutgoingWebhook2Form = observer((props: OutgoingWebhook2FormProps) => {
     [id]
   );
 
+  if (
+    (action === WebhookFormActionType.EDIT_SETTINGS || action === WebhookFormActionType.VIEW_LAST_RUN) &&
+    !outgoingWebhook2Store.items[id]
+  ) {
+    // nothing to show if we open invalid ID for edit/last_run
```
Comment (maskin25) [inline grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:79]: can it be unloaded in store yet at the moment this code is running?

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx [hunk 3]
```diff
@@ -49,35 +72,167 @@ const OutgoingWebhook2Form = observer((props: OutgoingWebhook2FormProps) => {
...
+    return null;
+  }
+
+  if (action === WebhookFormActionType.NEW || action === WebhookFormActionType.COPY) {
+    // show just the creation form, not the tabs
```
Comment (maskin25) [inline grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:84]: mabe just make LastRun tab disabled for those cases?

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx [hunk 4]
```diff
@@ -1,15 +1,20 @@
...
+import { useHistory } from 'react-router-dom';
 
 import GForm from 'components/GForm/GForm';
 import Text from 'components/Text/Text';
+import OutgoingWebhook2Status from 'containers/OutgoingWebhook2Status/OutgoingWebhook2Status';
 import { WithPermissionControlTooltip } from 'containers/WithPermissionControl/WithPermissionControlTooltip';
 import { OutgoingWebhook2 } from 'models/outgoing_webhook_2/outgoing_webhook_2.types';
+import { WebhookFormActionType } from 'pages/outgoing_webhooks_2/OutgoingWebhooks2.types';
 import { useStore } from 'state/useStore';
+import { KeyValuePair } from 'utils';
```
Comment (maskin25) [inline grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:15]: what the pros of using this data structure?

File: grafana-plugin/src/pages/integrations/Integrations.tsx [hunk 5]
```diff
@@ -447,9 +445,7 @@ class Integrations extends React.Component<IntegrationsProps, IntegrationsState>
           </div>
         )}
       >
-        {({ openMenu }) => (
-          <HamburgerMenu openMenu={openMenu} listBorder={ACTIONS_LIST_BORDER} listWidth={ACTIONS_LIST_WIDTH} />
-        )}
+        {({ openMenu }) => <HamburgerMenu openMenu={openMenu} listBorder={2} listWidth={200} />}
```
Comment (maskin25) [inline grafana-plugin/src/pages/integrations/Integrations.tsx:448]: to be precise it's a kebab (see https://alvarotrigo.com/blog/assets/imgs/2021-12-14/hamburger-menu-types-examples.webp) =))

**Review by maskin25 — APPROVED** [review]

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx (hunk 4 above)
Comment (teodosii) [reply grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:15]: It's a key-value structure for when you need to keep groups of key/value pairs. It's used in other places as well.

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx (hunk 3 above)
Comment (teodosii) [reply grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:84]: This follows the figma where on new/copy there's no tabs shown, plus the content is slightly different.

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx (hunk 1 above)
Comment (teodosii) [reply grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:58]: Agree, I'll refactor it

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx (hunk 2 above)
Comment (teodosii) [reply grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:79]: I'm not sure what you mean by this. This is when you're hitting an ID that was previously deleted or it just doesn't exist. There's no such object in the store.

**Comment by teodosii** [comment]
Will postpone adding the tests as I need to help Maxim add the template editor for webhooks.

File: grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx (hunk 2 above)
Comment (maskin25) [reply grafana-plugin/src/containers/OutgoingWebhook2Form/OutgoingWebhook2Form.tsx:85]: as I understand we don't show anything if there is no such item in the store, but in this case we need to force load it within this component

**Pull request merged**